RRT_star.py

```diff
@@ -37,119 +43,149 @@ def connection(self, va, vb):
         """Return the cost of connecting two verices"""
         return np.linalg.norm(va -vb)
 
-    def expand_graph(self, min_edge_len, max_edge_len, niu=0.8):
-        """Add a vertex to the RRT graph"""
-        q = self.workspace.sample()
-        v_nearest, dist = self.closest_vertex(q)
-        q_nearest = v_nearest.coord
+    
+    def query(
+        self,
+        start,
+        goal,
+        n=1000,
+        min_edge_len=0.5,
+        max_edge_len=1,
+        niu = 0.8
+    ):
+        new_size = self.graph.n 
+        start_time = time.time()
+        samples = []
+        prev = 0
+        while new_size<n:
+            # if prev !=new_size:
+            #     print(new_size)
+            #     prev = new_size
+            # TODO maybe change min and max edge len based on how many points have been taken?
+            # add radius with minimum path
+            if len(samples) == 0: 
```

Comment:
What is the different between the case when `len(samples) == 0` vs `len(samples) > 0`? Can you comment what is going on here?

Reply:
samples is a list that is actually used mostly in the unbounded RRT star although I included it here for consistency. It stores all the points that have been samples and are safe to add to the tree. When samples is empty the code looks for a new point that then tries to add. If samples is not empty then the workspace is not sampled as there are already points that can be added to the tree